src/ol/format/wfs/filter.js

```diff
@@ -0,0 +1,617 @@
+ * @param {string=} opt_singleChar pattern character which matches any single
+ *    string character. Default is '.'.
+ * @param {string=} opt_escapeChar Escape character which can be used to escape
+ *    the pattern characters. Default is '!'.
+ * @returns {!ol.format.wfs.filter.IsLike} `<PropertyIsLike>` operator.
+ * @api
+ */
+ol.format.wfs.filter.like = function(propertyName, pattern,
+    opt_wildCard, opt_singleChar, opt_escapeChar) {
+  return new ol.format.wfs.filter.IsLike(propertyName, pattern,
+    opt_wildCard, opt_singleChar, opt_escapeChar);
+};
+
+
+/**
+ * @classdesc
+ * Abstract class; normally only used for creating subclasses and not instantiated in apps.
+ * Base class for WFS GetFeature filters.
+ *
+ * @constructor
+ * @param {!string} tagName The XML tag name for this filter.
+ * @extends {ol.Object}
+ * @api
+ */
+ol.format.wfs.filter.Filter = function(tagName) {
+
+  goog.base(this);
+
+  /**
+   * @private
+   * @type {!string}
+   */
+  this.tagName_ = tagName;
+};
+goog.inherits(ol.format.wfs.filter.Filter, ol.Object);
+
+/**
+ * The XML tag name for a filter.
+ * @returns {!string} Name.
+ */
+ol.format.wfs.filter.Filter.prototype.getTagName = function() {
+  return this.tagName_;
+};
+
+
+// Logical filters
+
+
+/**
+ * @classdesc
+ * Abstract class; normally only used for creating subclasses and not instantiated in apps.
+ * Base class for WFS GetFeature logical filters.
+ *
+ * @constructor
+ * @param {!string} tagName The XML tag name for this filter.
+ * @extends {ol.format.wfs.filter.Filter}
+ */
+ol.format.wfs.filter.Logical = function(tagName) {
+  goog.base(this, tagName);
+};
+goog.inherits(ol.format.wfs.filter.Logical, ol.format.wfs.filter.Filter);
+
+
+/**
+ * @classdesc
+ * Abstract class; normally only used for creating subclasses and not instantiated in apps.
+ * Base class for WFS GetFeature binary logical filters.
+ *
+ * @constructor
+ * @param {!string} tagName The XML tag name for this filter.
+ * @param {!ol.format.wfs.filter.Filter} conditionA First filter condition.
+ * @param {!ol.format.wfs.filter.Filter} conditionB Second filter condition.
+ * @extends {ol.format.wfs.filter.Logical}
+ */
+ol.format.wfs.filter.LogicalBinary = function(tagName, conditionA, conditionB) {
+
+  goog.base(this, tagName);
+
+  /**
+   * @public
+   * @type {!ol.format.wfs.filter.Filter}
+   */
+  this.conditionA = conditionA;
+
+  /**
+   * @public
+   * @type {!ol.format.wfs.filter.Filter}
+   */
+  this.conditionB = conditionB;
+
+};
+goog.inherits(ol.format.wfs.filter.LogicalBinary, ol.format.wfs.filter.Logical);
+
+
+/**
+ * @classdesc
+ * Represents a logical `<And>` operator between two filter conditions.
+ *
+ * @constructor
+ * @param {!ol.format.wfs.filter.Filter} conditionA First filter condition.
+ * @param {!ol.format.wfs.filter.Filter} conditionB Second filter condition.
+ * @extends {ol.format.wfs.filter.LogicalBinary}
+ * @api
+ */
+ol.format.wfs.filter.And = function(conditionA, conditionB) {
```

Comment:
Is it really needed to implement this function if it only calls the parent? if it for jsdoc ?